### DISCUSSION TITLE
How to create a request/response pipeline with Annotations Framework

### DISCUSSION THREAD

**Discussion by petro2050** [aws/aws-lambda-dotnet#1629] [Q&A]
I'm just getting started with Annotations, and I see it is possible to do dependency injection. I'm curious about how we can emulate something like ASP.NET Core Middleware to implement global services like authentication/authorization, rate limiting, request validation, logging, etc. and pipeline them.

Also, is it possible to pass a complex request object to a lambda handler? The examples I see demonstrate basic ops like adding two numbers.

**Comment by ashishdhingra** [comment]
@petro2050 Please refer the following for guidance:
- https://github.com/aws/aws-lambda-dotnet/tree/master/Libraries/src/Amazon.Lambda.Annotations#dependency-injection-integration
- https://www.youtube.com/watch?v=FPaznlIY02s
